examples/remote_procedure/toxicity_detection/webapp/src/components/core_components.jsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import React, { Fragment, useState } from "react";
+
+function LoadingScreen() {
+  return <Directions>Loading...</Directions>;
+}
+
+function Directions({ children }) {
+  return (
+    <section className="hero is-light">
+      <div className="hero-body">
+        <div className="container">
+          <p className="subtitle is-5">{children}</p>
+        </div>
+      </div>
+    </section>
+  );
+}
+function Instructions() {
+  return (
+    <div>
+      <h1>Toxicity Detection Model</h1>
+      <p>
+        To submit this task, you'll need to enter a sentence of sentences in the
+        input box below. The model will calculate the toxicity of the inputted
+        text.
+      </p>
+    </div>
+  );
+}
+
+function TaskFrontend({ handleSubmit, handleToxicityCalculation }) {
+  const [text, setText] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [result, setResult] = useState("");
+  const [toxicity, setToxicity] = useState(0);
+
+  function calculateToxicity() {
+    setIsLoading(true);
+    handleToxicityCalculation({
+      text: text,
+    }).then((response) => {
+      setIsLoading(false);
+      const parsedToxicity = parseFloat(response.toxicity);
+      setToxicity(parsedToxicity);
+      if (parsedToxicity <= 0.5) {
+        handleSubmit({ toxicity: response.toxicity });
+      } else {
+        setResult(
+          `The statement, "${text}," has a toxicity of: ${response.toxicity}. This message is too toxic to submit.`
+        );
+      }
+    });
+  }
```

Comment:
Good idea 👍 . Just added it

Reply:
It would be nice to have some non-dev oriented visual UI error messaging for anyone using this task. Could be something simple just saying "Something went wrong."

(*Sidenote: this is also a case where the Feedback component we build will be useful - you can imagine someone leaving feedback like: "hey it seems like your task is broken! the toxicity result never loads!"*)

A `console.error` is only really useful for devs, and even then it's not super intuitive to have to check the console, without any indication from the UI that something went wrong.